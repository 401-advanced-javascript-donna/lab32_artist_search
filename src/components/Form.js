import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ handleSubmit, handleChange, search }) => {
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" name="search" onChange={handleChange} value={search} />
      <button>Go!</button>
    </form>
  );
};

Form.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  search: PropTypes.string.isRequired
};

export default Form;
