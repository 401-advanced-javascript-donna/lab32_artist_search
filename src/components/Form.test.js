import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form handleChange={()=>{}} handleSubmit={()=>{}} search='' />);
    expect(wrapper).toMatchSnapshot();
  });
});
