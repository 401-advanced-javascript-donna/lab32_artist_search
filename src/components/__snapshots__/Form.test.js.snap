// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="search"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button>
    Go!
  </button>
</form>
`;
